shm/src/player.cpp

```diff
@@ -1 +1,15 @@
 #include "player.hpp"
+
+#include <algorithm>
+
+size_t Player::calculateavailableSpace() {
+    size_t reservedSpace = 0;
+    std::for_each(ship_->getallCargo().begin(),
+                  ship_->getallCargo().end(),
+                  [&reservedSpace](const Cargo& cargo) {
+                      reservedSpace += cargo.getAmount();
+                  });
+    size_t freeSpace = ship_->getCapacity() - reservedSpace;
+
+    return freeSpace;
```

Comment:
What do you think about:

```suggestion
size_t reservedSpace = std::accumulate(
    ship_->getAllCargo().begin(),
    ship_->getAllCargo().end(),
    0,
    [](size_t space, const Cargo& cargo) {
        return space += cargo.getAmount();
    });
size_t availableSpace = ship_->getCapacity() - reservedSpace;
```

Sorry for formatting :)